core/job/service/job_service.go

```diff
@@ -71,9 +71,9 @@ type Engine interface {
 }
 
 type PluginService interface {
-	Info(context.Context, job.TaskName) (*plugin.Info, error)
-	GenerateDestination(ctx context.Context, taskName job.TaskName, configs map[string]string) (job.ResourceURN, error)
-	GenerateDependencies(ctx context.Context, taskName job.TaskName, svcAcc, query string, destinationURN job.ResourceURN) ([]job.ResourceURN, error)
+	Info(ctx context.Context, taskName string) (*plugin.Info, error)
+	GenerateUpstreams(ctx context.Context, taskName string, config, assets map[string]string) (resourceURNs []string, err error)
```

Comment:
Can we rename this method to `IdentifyUpstreams` or something like that, as we are just trying to identify and not actually generate.